enumerables.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1 
+    end
+    self
+  end
+
+  def my_each_whit_index
+    i = 0
+    while i < size 
+      yield(self[i], i)
+      i += 1
+    self
+    end
+  end
+
+  def my_select 
+    result = []
+    my_each {|item| return.push(item) if yield (item) }
+    result 
+  end
+
+  def my_all?
+    my_each {|item| return false if yield(item)}
```

Comment:
my_all method has three conditions; when a block is given, when a pattern is given and when no pattern or block is given.

However, use this code snippet as a guide.

```  
def my_all?(pattern = nil)
    result = true
    if block_given?
      my_each { |ele| result &= (yield ele) }
    elsif pattern
      my_each { |ele| result &= pattern === ele }
    else
      my_each { |ele| result &= ele }
    end
    result
  end 
```